js/components/Message/BodyPieces.js

```diff
@@ -8,10 +8,15 @@ BodyPieces.propTypes = {
 function BodyPieces(props) {
   //@TODO add support for styles.
 
-  let pieces = props.pieces.map(function(piece) {
+  let pieces = props.pieces.map(function(piece, i) {
     const separator = " ";
+    // Body pieces don't ever change order so we can use the index
```

Comment:
Yeah, if there's never any way the order will change (which sounds right to me), then array indexes should be fine. Here's a discussion of keys that touches on this: https://github.com/facebook/react/issues/1342#issuecomment-39230939%22%20rel=%22nofollow